model-engine/model_engine_server/infra/repositories/live_tokenizer_repository.py

```diff
@@ -0,0 +1,149 @@
+import os
+from collections import namedtuple
+from functools import lru_cache
+from typing import Dict, Optional
+
+from huggingface_hub import list_repo_refs
+from huggingface_hub.utils._errors import RepositoryNotFoundError
+from model_engine_server.core.loggers import logger_name, make_logger
+from model_engine_server.domain.exceptions import ObjectNotFoundException
+from model_engine_server.domain.gateways.llm_artifact_gateway import LLMArtifactGateway
+from model_engine_server.domain.repositories.tokenizer_repository import TokenizerRepository
+from transformers import AutoTokenizer
+
+logger = make_logger(logger_name())
+
+
+TOKENIZER_FILES_REQUIRED = [
+    "tokenizer.json",
+    "tokenizer_config.json",
+    "special_tokens_map.json",
+]
+TOKENIZER_FILES_OPTIONAL = [
+    "tokenizer.model",
+]
+TOKENIZER_TARGET_DIR = "/opt/.cache/model_engine_server/tokenizers"
+
+
+ModelInfo = namedtuple("ModelInfo", ["hf_repo", "s3_repo"])
```

Comment:
Would a Union type be more appropriate, given that it seems that the s3 thing is always None it seems?

Also might want to hide this too.

Reply:
dependency injection passes in non none where applicable